### DISCUSSION TITLE
Waves v12 plugins don't load from previous Waves v11/DAW session

### DISCUSSION THREAD

**Discussion by jaacMusic** [apohl79/audiogridder#275] [Q&A] [answered]
I found out that Waves v12 settings are not loading after upgrading from v11 to v12. Plugin monitor keeps opening and closing. I assuming plugins are trying to load but are not getting loaded. Plugins appear in AG plugin as gray out. See screenshot below
[AudioGridderPlugin_2020-12-19_11-13-33.zip](https://github.com/apohl79/audiogridder/files/5719766/AudioGridderPlugin_2020-12-19_11-13-33.zip)
[AudioGridderServer_2020-12-19_11-15-05.zip](https://github.com/apohl79/audiogridder/files/5719770/AudioGridderServer_2020-12-19_11-15-05.zip)

![image](https://user-images.githubusercontent.com/68957086/102693805-23486880-41eb-11eb-94ac-1ca489daaabc.png)

**Comment by apohl79** [comment, marked as answer]
The problem are these messages, where the server complains about plugins it does not find:

[11:17:13.652|worker|20a011cf4d8|client:dc2f3a20] adding plugin VST-SSLGChannel Stereo-838cb03c-53434753...
[11:17:13.668|agprocessor|7ff6d7e8b798] failed to find plugin descriptor

One thing I changed in beta10 is the use of plugin ID's. AG tries to use only the unique plugin ID and avoid using the disk location (which is the JUCE behavior). This works only, if there is no conflicting ID's. Once a collision is found (two plugins with the same ID), AG falls back to the JUCE default. This fallback decision remains going forward. AG stores a settings value for it (see below). This is the case for you, as you have multiple plugins with different disk locations, but the same plugin ID. This happens for example when upgrading Waves container versions, when the old and new containers are installed.

Example from your log:
[11:15:07.764|server|20a62a41ce8]   SSLGChannel Stereo [VST-SSLGChannel Stereo-**bf580957**-53434753] format=VST ins=1 outs=1 instrument=0
[11:15:07.764|server|20a62a41ce8]   SSLGChannel Stereo [VST-SSLGChannel Stereo-**eeb8343c**-53434753] format=VST ins=1 outs=1 instrument=0

As you see, two SSLGChannel plugins, same plugin ID, different disk locations. 

Now, the ID AG tries to load is: "VST-SSLGChannel Stereo-**838cb03c**-53434753", which has a different disk location hash than the other two. Thats why the server fails (and the plugin stays greyed out).

Waves is using plugin file names like "WaveShell#number#-VST #container version#_x64.dll" for all of their plugins. I guess the number part depends only on the installation order of plugins and is thus not reliable.

I think you need to figure out a way to "clean up" your setup. Try having only one container version of waves plugins installed. After this, you have to make AG server try to use disk location independent ID's again, by manually updating the config, as this setting has no UI:

1) stop the server
2) open the server config in notepad (see https://audiogridder.com/server/#config)
3) Update the line: 
`"UseJucePluginIDs": true,`
to
`"UseJucePluginIDs": false,`

After starting the server, check the server log. The SSLGChannel log entries from above should look like this now:

[11:15:07.764|server|20a62a41ce8]   SSLGChannel Stereo [VST-SSLGChannel Stereo-53434753] format=VST ins=1 outs=1 instrument=0

Just with a single hashed value. Also make sure, there is NO single line with the warning string "!ID collision", as once you see it, AG falls back to JUCE ID's as described above.

Now AG uses ID's that potentially work across servers and across operating systems as long as the plugin ID's and names are matching.

The problem that remains is that the AG plugin stores the plugin ID's of loaded remote plugins and will continue to use the old ID's. There is no way (yet) to migrate existing projects to the new ID's automatically. You basically have to re-insert all plugins and will loose your settings unfortunately. I think that will be a useful feature going forward, but doesn't help you right now.

Ok... long answer, hope you get it sorted!

**Comment by jaacMusic** [comment]
I see been an issue every time I upgrade to a newer version of a plugin.
Is it possible to go by plugin name and disregard hash/ids. It would be rare for different vst developers to use the same name; moreover, we could use the company name + plugin name as a unique identifier. What do you think?

**Comment by apohl79** [comment]
The name is already part of the ID string to avoid ID collisions in case multiple plugins use the same unique ID. As said, the disk location is only used when there are collisions (multiple plugins with an identical ID string). Only than the hashed disk location is added. And this would be the case if you have V11 and V12 versions of the same waves plugin installed.

**Comment by jaacMusic** [comment]
Sorry if I didn't clarify it well. I have two AG servers (one is a backup of the other). I upgraded my backup to Beta 10 but I didn't upgrade Waves v11 just in case v12 will break my session. I tested with my backup beta 10 and my saved session doesn't load the plugins, they stay grayed out. Again, this backup doesn't have any other version of Waves but I still get the same issue.
Hope I made it a bit more clear. Thanks!!

**Comment by apohl79** [comment]
And the two servers run on different computers or the same?

**Comment by jaacMusic** [comment]
Different computers, both have windows 10

**Reply by apohl79** [reply]
Ok, I think if you manage to resolve the ID conflicts as described above on both servers and make AG use only the plugin name/id, you should be able to address the waves plugins independently of their version by just switching between the servers.

**Comment by jaacMusic** [comment]
I'm trying to figure out how to edit VST ID on old session that has an older version of the VST. Is the VST ID autogenerate or taken from the VST ID. I was trying to find in the Windows registry or on file something that matches and/or that I could change.

By the way, I opened a Cakewalk session file in notepad and found entries like this: ""activeServerStr":"DSP","loadedPlugins":[["VST3-API-550A Stereo-fe8af03e-fb2c5297" Sadly I can't edit this file in notepad to update every VST id with the new ID.

It seems VST IDs are installation-specific. I have various sessions that would not load in the backup or production server because they have different IDs. Do you think it would be feasible to have an option to override VST ID from the AG plugin side? That's why I was asking to use the company and plugin name instead of ID. Any thoughts are welcome! Thanks again!!!

**Comment by CaBuffalo** [comment]
Just want to confirm here that Waves V12 works on Windows 10 server and Logic configuration.

**Reply by CaBuffalo** [reply]
Here is a side note. I installed the plugins, restart my PC as it's required. I found that I can't load any plugins after the server rescanned the new plugins. Then I restarted my PC again. Then everything works.

**Comment by jaacMusic** [comment]
Andres, I did various tests where I upgraded my Waves version and I couldn't get any of my sessions that use the old Waves version to work. I even tried a clean upgrade (uninstalled everything and remove registry and leftover files from Windows 10) and that didn't work. Tried on both my main server and backup server and I couldn't get this to work.
Tracks that had Waves plugins loaded that were loaded directly on the track (not using AG plugin) worked fine when upgraded the Wave plugins on my DAW system.
I know you have mentioned that AG uses VST installation IDs and I could see where Waves installations will change these. Just wanted to see if there is a workaround to this as upgrading my Waves (possibly other plugins) could cause me to lose all my configuration.

**Comment by CaBuffalo** [comment]
I think this is the DAW that gives you a headache. I did the same thing in
Logic and it works fine if I could remember.

Let's see what Andreas says anyways. :)

On Thu, Jan 14, 2021 at 7:05 PM jaacMusic <notifications@github.com> wrote:

> Andres, I did various tests where I upgraded my Waves version and I
> couldn't get any of my sessions that use the old Waves version to work. I
> even tried a clean upgrade (uninstalled everything and remove registry and
> leftover files from Windows 10) and that didn't work. Tried on both my main
> server and backup server and I couldn't get this to work.
> Tracks that had Waves plugins loaded that were loaded directly on the
> track (not using AG plugin) worked fine when upgraded the Wave plugins on
> my DAW system.
> I know you have mentioned that AG uses VST installation IDs and I could
> see where Waves installations will change these. Just wanted to see if
> there is a workaround to this as upgrading my Waves (possibly other
> plugins) could cause me to lose all my configuration.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/275#discussioncomment-283408>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6AWAHKV2A2XRCLLX2TSZ6WJLANCNFSM4VCMUNDQ>
> .
>


-- 
Kevin Ma
Supported by Gmail.

**Comment by apohl79** [comment]
I will see if I can increase the backwards compatibility. I totally understand your pain.

**Reply by jaacMusic** [reply]
Thanks again Andres for giving this an additional look!

**Reply by apohl79** [reply]
Please try beta11. The errors from the original logs in this thread should be solved now. I hope this solves your problems.

**Reply by jaacMusic** [reply]
Great! I'll test it out tonight after work. Thanks!!

**Reply by jaacMusic** [reply]
I didn't get much time to play with but so far my old session is now loading without issues! Thanks again! I'll try to do some additional testing this weekend.
